rules/exec_properties.bzl

```diff
@@ -0,0 +1,134 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This file contains macros to create and manipulate dictionaries of properties to be used as execution properties for RBE.
+"""
+
+def _add(dict,
+         var_name,
+         key,
+         value,
+         verifier_fcn=None,
+         transform_fcn=None,
+        ):
+    """Add a key-value to a dict.
+    
+    If value is None, don't add anything.
+    The dict will always be a string->string dict, but the value argument to this function may be of a different type.
+    
+    dict: The dict to update.
+    var_name: Used for error messages.
+    key: The key in the dict.
+    value: The value provided by the caller. This may or may not be what ends up in the dict.
+    verifier_fcn: Verifies the validity of value. On error, it's the verifier's responsibility to call fail().
+    transform_fcn: Transform the value provided to this function with a value to put in the dict.
+                   Note that a transform_fcn is not needed for casting a non string to string, this will be done anyway.
+    """
+    if value == None:
+        return
+    if verifier_fcn != None:
+        verifier_fcn(var_name, value) # verifier_fcn will fail() if necessary
+    if transform_fcn != None:
+        value = transform_fcn(value)
+    dict[key] = str(value)
+
+
+def _verify_string(var_name, value):
+    if type(value) != "string":
+        fail("%s must be a string" % var_name)
+
+
+def _verify_bool(var_name, value):
+    if type(value) != "bool":
+        fail("%s must be a bool" % var_name)
+
+def _verify_os(var_name, value):
+    _verify_string(var_name, value)
+    valid_os_list = ["Linux", "Windows"]
+    if value not in valid_os_list:
+        fail("%s must be one of %s" % (var_name, valid_os_list))
+
+
+def _transform_network(value):
+    return "standard" if value else "off"
+
+
+def create_exec_properties_dict(
+    # The asterisk means that all following arguments must be specified by name at the call site.
+    # This allows future insertions of arguments to this function without breaking callers which might have otherwise depended on argument ordering.
+    *,
+    container_image=None,
+    pool=None,
+    os_family=None,
+    gce_machine_type=None,
+    jdk_version=None,
+    docker_add_capabilities=None,
+    docker_drop_capabilities=None,
+    docker_network_enabled=None,
+    docker_privileged=None,
+    docker_run_as_root=None,
+    docker_runtime=None,
+    docker_sibling_containers=None,
+    docker_ulimits=None,
+    docker_use_urandom=None):
+    """Return a dict with exec_properties that are supported by RBE.
+    
+    Args:
+    container_image (string): The container that RBE will execute in.
+    pool (string): If specified, the action will be executed only on this worker pool.
```

Comment:
so I'm not sure using "the action will be" is clear here as this macro does not actually run any actions. How about:
"the dict returned will include a property that restricts execution on the given worker pool"
same for all descriptions below

Reply:
Now that I think about it, I think it might be better to just remove these comments explaining what each argument is and instead just point to https://cloud.google.com/remote-build-execution/docs/remote-execution-environment#remote_execution_properties since each of these arguments maps to an argument that is documented there.

Reply:
I removed these comments.